### DISCUSSION TITLE
LockX, LockY, LockRelativeX, LockRelativeY, DisableWorldMotion, and CanBePushed should be a setting when saving a submarine

### DISCUSSION THREAD

**Discussion by girlyguppy** [FakeFishGames/Barotrauma#15084] [Ideas and suggestions]
These settings would open up a lot of possibilities in the submarine editor, particularly for mapmaking but really just for anything.

LockX and LockY is obvious. when testing a beacon or outpost in editor, you may want to have this on by default, otherwise they float.
when making elevators for beacons or outposts, you usually want them to move on only 1 axis. trying to add rails or use dockingport jank is painful, and this would simplify it tenfold.

LockRelativeX and LockRelativeY would come in handy for elevators on moving submarines, as outright locking the *global* x or y of the shuttle would prevent the mainsub from moving, and cause lots of issues. Locking the relative coordinate would simply make it so that when the submarine is moving, the moving shuttles would not behave strangely. (for example, going up slower due to the sub rising)
Some submarines have retractable turrets. they basically HAVE to use the dockingport method because otherwise moving the submarine would result in them sliding around, which sucks, as the intent is obviously to have a rigid connection.

DisableWorldMotion would be a setting for shuttles that would allow it to move relative to the submarine, but otherwise feel "attached" when the shuttle is stationary. This could be used to solve several other issues that shuttle-related technology has to fight with. For example, imagine a submarine that had something orbiting it. This should also be accounted for on navigation terminals, so that the "maintain position" becomes relative to the mainsub, instead of to the shuttle. however it must be noted that this bug must be fixed first: https://github.com/FakeFishGames/Barotrauma/discussions/14886

CanBePushed would essentially mean it's impervious to being moved under any power aside from its own. e.g. the engine and ballasts can move it, but it would be able to resist being slammed into by an endworm.

My final suggestion is to add winches. Essentially a permanent, controllable version of the railgunharpoon, it would allow you to move a shuttle/elevator up and down (or side-to-side) at will. Perhaps a "strength" and "maxlength" property, which would control how much it could lift (e.g it would stop being able to winch upwards when full of water), and the "maxlength" would simply control how long the connecting string can go. "canbreak" would be another possibility, maybe for it to break and be repaired/recovered mid-round. 

Adding more moving parts to the game is always very difficult or jank, and these suggestions would make them much more realistic and doable.

**Comment by AziKarnage** [comment]
Need this fr

**Comment by jigardi** [comment]
@girlyguppy You've probably already been notified, but the big man himself replied that #14886 has been fixed!
